src/Product/WCProductAdapter.php

```diff
@@ -941,7 +941,8 @@ protected function map_gtin(): WCProductAdapter {
 			return $this;
 		}
 
-		$global_unique_id = $this->wc_product->get_global_unique_id();
+		// avoid dashes and other unsupported format
+		$global_unique_id = preg_replace( '/[^0-9]/', '', $this->wc_product->get_global_unique_id() );
```

Comment:
Unfortunate that it would look odd to use the WC function `prepare_gtin` here. But it's probably better since it's not available in all versions of WC that support GTIN.